src/pykeen/triples/triples_factory.py

```diff
@@ -146,59 +146,465 @@ def _ensure_ids(
+        Create a triples factory without any label information.
 
-    #: A vectorized version of relation_label_to_id; initialized automatically
-    _vectorized_relation_mapper: Callable[..., np.ndarray] = dataclasses.field(init=False)
+        :param mapped_triples: shape: (n, 3)
+            The ID-based triples.
+        :param num_entities:
+            The number of entities. If not given, inferred from mapped_triples.
+        :param num_relations:
+            The number of relations. If not given, inferred from mapped_triples.
+        :param create_inverse_triples:
+            Whether to create inverse triples.
+        :param metadata:
+            Additional metadata to store in the factory.
 
-    #: A vectorized version of entity_id_to_label; initialized automatically
-    _vectorized_entity_labeler: Callable[..., np.ndarray] = dataclasses.field(init=False)
+        :return:
+            A new triples factory.
+        """
+        if num_entities is None:
+            num_entities = mapped_triples[:, [0, 2]].max().item() + 1
+        if num_relations is None:
+            num_relations = mapped_triples[:, 1].max().item() + 1
+        return CoreTriplesFactory(
+            mapped_triples=mapped_triples,
+            num_entities=num_entities,
+            num_relations=num_relations,
+            create_inverse_triples=create_inverse_triples,
+            metadata=metadata,
+        )
+
+    @property
+    def num_entities(self) -> int:  # noqa: D401
+        """The number of unique entities."""
+        return self._num_entities
+
+    @property
+    def num_relations(self) -> int:  # noqa: D401
+        """The number of unique relations."""
+        if self.create_inverse_triples:
+            return 2 * self.real_num_relations
+        return self.real_num_relations
+
+    @property
+    def real_num_relations(self) -> int:  # noqa: D401
+        """The number of relations without inverse relations."""
+        return self._num_relations
+
+    @property
+    def num_triples(self) -> int:  # noqa: D401
+        """The number of triples."""
+        return self.mapped_triples.shape[0]
+
+    def extra_repr(self) -> str:
+        """Extra representation string."""
+        d = [
+            ('num_entities', self.num_entities),
+            ('num_relations', self.num_relations),
+            ('num_triples', self.num_triples),
+            ('inverse_triples', self.create_inverse_triples),
+        ]
+        d.extend(sorted(self.metadata.items()))  # type: ignore
+        return ', '.join(
+            f'{k}="{v}"' if isinstance(v, str) else f'{k}={v}'
+            for k, v in d
+        )
+
+    def __repr__(self):  # noqa: D105
+        return f'{self.__class__.__name__}({self.extra_repr()})'
+
+    def with_labels(
+        self,
+        entity_to_id: Mapping[str, int],
+        relation_to_id: Mapping[str, int],
+    ) -> "TriplesFactory":
+        """Add labeling to the TriplesFactory."""
+        # TODO: Check labelings
```

Comment:
can you elaborate on what needs to be done here? We can push it to another PR but at least write some more so we don't forget

Reply:
The mappings should be compatible to the triples, i.e. their value range cover at least all occurring entity/relation IDs.

Reply:
It should be an easy fix:

```python
set(self.mapped_triples[:, [0, 2]].unique().tolist()).issubset(entity_to_id.values())
```

Reply:
Then let's put it in this PR

Reply:
Done in 6054267